src/main/java/carpet/script/value/EntityValue.java

```diff
@@ -891,21 +892,49 @@ else if (v instanceof ListValue)
             throw new InternalExpressionException("'effect' needs either no arguments (clear) or effect name, duration, and optional amplifier, show particles and show icon");
         });
 
-        // gamemode
-        // spectate
-        // "fire"
-        // "extinguish"
-        // "silent"
-        // "gravity"
-        // "invulnerable"
-        // "dimension"
-        // "item"
-        // "count",
-        // "age",
-        // "effect_"name
-        // "hold"
-        // "hold_offhand"
-        // "jump"
+        put("gamemode", (e,v)->{
+            if(!(e instanceof ServerPlayerEntity)){
+                throw new InternalExpressionException("Cannot change the gamemode of non-player entities");
```

Comment:
API should not throw exceptions if the arguments types and counts are fine, but are not applicable - should return null. Exeptions is when fromat of arguments is incorrect.

Reply:
done